api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/EriDSDCohortQueries.java

```diff
@@ -1370,6 +1372,124 @@ public CohortDefinition getBreastfeedingPatientsStartingART() {
     return cd;
   }
 
+  @DocumentedDefinition(
+      "Number of active patients on ART (Non-pregnant and Non-Breastfeeding not on TB treatment) who are in CA")
+  public CohortDefinition getPatientsWhoAreActiveAndParticipatingInAccessionClubs() {
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+    cd.addParameter(new Parameter("startDate", "Start Date", Date.class));
+    cd.addParameter(new Parameter("endDate", "End Date", Date.class));
+    cd.addParameter(new Parameter("location", "Location", Location.class));
+
+    CohortDefinition txCurr = txCurrCohortQueries.getTxCurrCompositionCohort("TX_CURR", true);
+    CohortDefinition startOrContinueCA = getPatientsWithStartOrContinueCA();
+
+    String mappings = "onOrBefore=${endDate},location=${location}";
+    cd.addSearch("txCurr", EptsReportUtils.map(txCurr, mappings));
+
+    String caMappings = "onOrAfter=${startDate},onOrBefore=${endDate},locationList=${location}";
+    cd.addSearch("startOrContinueCA", EptsReportUtils.map(startOrContinueCA, caMappings));
+
+    cd.setCompositionString("txCurr AND startOrContinueCA");
+
+    return cd;
+  }
```

Comment:
Please include the filter for patients who are **"Non-pregnant & non-breastfeeding and not in TB treatment"** here.  Otherwise, line 536 in this class file **"api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/EriDSDDataset.java"** will not return the correct value

> NB:  Remember to include all 3 filters (Non-pregnant, non-breastfeeding and not-in-TB -treatment).

Reply:
done